### PR TITLE
feat: attendee dialog

### PR DESCRIPTION
**Pull request by juandgaines**
add attendee

### CODE REVIEW

File: agenda/data/src/main/java/com/juandgaines/agenda/data/event/DefaultAttendeeRepository.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.juandgaines.core.data.network.safeCall
+import com.juandgaines.core.domain.util.DataError
+import com.juandgaines.core.domain.util.Result
+import com.juandgaines.core.domain.util.map
+import kotlinx.coroutines.CoroutineScope
+import javax.inject.Inject
+
+class DefaultAttendeeRepository @Inject constructor(
+    private val attendeeApi: AttendeeApi,
+    private val applicationScope: CoroutineScope,
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/juandgaines/agenda/data/event/DefaultAttendeeRepository.kt:16]: `applicationScope` is not used in this class atm 👀

File: agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/Attendee.kt [hunk 2]
```diff
@@ -2,12 +2,28 @@ package com.juandgaines.agenda.domain.agenda
...
 
+interface IAttendee{
+    val email: String
+    val fullName: String
+    val userId: String
+    val isGoing: Boolean
+    val isUserCreator: Boolean
+}
+
 data class Attendee(
```
Comment (Kai-Cilliers) [inline agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/Attendee.kt:13]: What is your thinking behind an interface here? Besides being able to place their implementations in the same list, I don’t see much benefit. An interface also leaves room for future implementations beyond the original intention of this approach.

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemScreen.kt [hunk 3]
```diff
@@ -275,6 +288,31 @@ fun AgendaItemScreen(
...
+                        onDismiss = {
+                            onAction(AgendaItemAction.DismissAttendeeDialog)
+                        },
+                        textState = state.details.attendeeEmailBuffer,
+                        isError = state.details.isEmailError,
+                        primaryButton = {
+                            TaskyActionButton(
+                                text = stringResource(id = R.string.add),
+                                isLoading = state.details.isAddingVisitor,
+                                enabled = state.details.attendeeEmailBuffer.text.isNotEmpty() && !
```
Comment (Kai-Cilliers) [inline agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemScreen.kt:305]: The requirements doc states that a valid email is required.

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt [hunk 4]
```diff
@@ -442,6 +462,119 @@ class AgendaItemViewModel @Inject constructor(
...
+                    updateState { state->
+                        state.copy(
+                            details = updateDetailsIfEvent { d->
+                                d.copy(
+                                    isAddingVisitor = true
+                                )
+                            }
+                        )
+                    }
+                    if (emailPatterValidator.matches(action.email)){
```
Comment (Kai-Cilliers) [inline agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt:477]: Ah, I see you validate the email here. Technically, the requirements do state that the “add” button is only enabled when the email is valid. For Tasky, you don’t need to make changes to fix this, but when working for clients, these small details matter. That is why you need to clarify the requirements doc with them before you start development.

File: agenda/data/src/main/java/com/juandgaines/agenda/data/event/DefaultAttendeeRepository.kt (hunk 1 above)
Comment (juandgaines) [reply agenda/data/src/main/java/com/juandgaines/agenda/data/event/DefaultAttendeeRepository.kt:16]: done

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt (hunk 4 above)
Comment (juandgaines) [reply agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt:477]: got it I will change it

**Pull request merged**